src/engine/QueryPlanner.h

```diff
@@ -165,6 +165,9 @@ class QueryPlanner {
     size_t getSizeEstimate() const;
   };
 
+  using FiltersAndOptionalSubstitutes =
+      std::vector<std::pair<SparqlFilter, std::optional<SubtreePlan>>>;
```

Comment:
Consider not using `pair`, but a properly named struct with member functions etc.